libr/anal/sign.c

```diff
@@ -1045,6 +1043,190 @@ static double matchGraph(RSignItem *a, RSignItem *b) {
 	return total / 5.0;
 }
 
+static RSignItem *create_graph_sign_from_fcn(RAnal *a, RAnalFunction *fcn) {
+	r_return_val_if_fail (a && fcn, false);
+
+	RSignGraph *graph = R_NEW0 (RSignGraph);
+	if (!graph) {
+		return NULL;
+	}
+	graph->cc = r_anal_function_complexity (fcn),
+	graph->nbbs = r_list_length (fcn->bbs);
+	graph->edges = r_anal_function_count_edges (fcn, &graph->ebbs);
+	graph->bbsum = r_anal_function_realsize (fcn);
+
+	char *name = r_str_new (fcn->name);
+	if (!name) {
+		free (graph);
+		return NULL;
+	}
+
+	RSignItem *item = r_sign_item_new ();
+	if (!item) {
+		free (name);
+		free (graph);
+		return NULL;
+	}
+
+	item->space = r_spaces_current (&a->zign_spaces);
+	item->name = name;
+	item->graph = graph;
+
+	return item;
+}
+
+typedef struct {
+	char *name;
+	double score;
+} _close_matches;
+
+static int score_cmpr(const void *a, const void *b) {
+	double sa = ((_close_matches *)a)->score;
+	double sb = ((_close_matches *)b)->score;
+
+	if (sa < sb) {
+		return 1;
+	}
+	if (sa > sb) {
+		return -1;
+	}
+	return 0;
+}
+
+static bool add_bestrow(RList *list, char *name, double score) {
+	_close_matches *row = R_NEW (_close_matches);
+	if (!row) {
+		return false;
+	}
+	row->name = strdup (name);
+	if (!row->name) {
+		free (row);
+		return false;
+	}
+	row->score = score;
+	r_list_append (list, row);
+	r_list_sort (list, &score_cmpr);
+	return true;
+}
+
+typedef struct {
+	RSignItem *test;
+	RList *output;
+	int count, index;
```

Comment:
`index` does not seem to be used, it can be removed

Reply:
ea9c97669 fixed